Assets/_BForBoss/_Weapons/Scripts/Weapons/AutomaticWeaponBehaviour.cs

```diff
@@ -1,17 +1,32 @@
+using FMODUnity;
 using UnityEngine;
 using UnityEngine.InputSystem;
 
 namespace Perigon.Weapons
 {
+    [RequireComponent(typeof(StudioEventEmitter))]
     public class AutomaticWeaponBehaviour : WeaponBehaviour
     {
+
+        private const string FIRE_RATE_PARAM = "FireRate";
+        private StudioEventEmitter _weaponFiringAudio;
+
+        private int _shotsFired;
+        
+        protected override void Awake()
+        {
+            base.Awake();
+            _weaponFiringAudio = GetComponent<StudioEventEmitter>();
+        }
+        
         protected override void OnFireInputAction(bool isFiring)
         {
             _isFiring = isFiring;
             
             if (!_isFiring)
             {
                 _timeSinceFire = 0;
+                _shotsFired = 0;
```

Comment:
So `_timeSinceFire` is set to 0 when `WeaponBehaviour.Reset()` is [called](https://github.com/PerigonGames/BForBoss/blob/ca58f30fe50539115d5fdf7c48e001cb0f13b1ed/Assets/_BForBoss/_Weapons/Scripts/Weapons/WeaponBehaviour.cs#L66), should we make that method virtual and override it here with `_shotsFired = 0`?

Reply:
Ah good catch, I'll do that!